src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/TdsParserStateObjectManaged.cs

```diff
@@ -72,13 +72,13 @@ internal override void AssignPendingDNSInfo(string userProtocol, string DNSCache
         internal void ReadAsyncCallback(SNIPacket packet, uint error)
         {
             ReadAsyncCallback(IntPtr.Zero, PacketHandle.FromManagedPacket(packet), error);
-            _sessionHandle.ReturnPacket(packet);
+            _sessionHandle?.ReturnPacket(packet);
         }
 
         internal void WriteAsyncCallback(SNIPacket packet, uint sniError)
         {
             WriteAsyncCallback(IntPtr.Zero, PacketHandle.FromManagedPacket(packet), sniError);
-            _sessionHandle.ReturnPacket(packet);
+            _sessionHandle?.ReturnPacket(packet);
```

Comment:
There's another issue in driver where this sessionHandle is null here, performing null check also fixes "Object reference not found" errors. But these sessionHandles are leaking data, which also need to be fixed. TBD.